### DISCUSSION TITLE
todo.txt how to hide hidden tasks containing a "h:1"

### DISCUSSION THREAD

**Discussion by aisbergde** [gsantner/markor#1591] [Ideas] [answered]
some other android todo.txt clients allow to define hidden tasks using a "h:1" somewhere in the row. Is there a way to toggle to hide or display them in Markor? This would allow to use this feature in the same files in different apps.

The rows are hidden because they contain some possible tags and context. Markor has another way to define them, but this works only in Markor.

Especially I mean the apps Simpletask and Mindstream.

**Comment by gsantner** [comment, marked as answer]
If you dont't want to see stuff, move it to other file, usually done.txt.

Otherwise, you don't, there is no such hiding magic. You edit a file in text editor and whats there in the file is fully shown to you.

**Comment by aisbergde** [comment]
I want to use the same file with Markor and with other todo.txt apps.

If you search for "todo.txt h:1" then you will find, that this is a quite commen syntax extension to use hidden tasks.

But these are not real tasks, they are just helpers. They are only used to configure the todo.txt in these apps. They use a different way for configuration than Markor. Markor also can add tags and context to be used, but they will be added only in Markor.

That's why these rows can be hidden in other Apps.

And I also want to hide them in Markor, and I can't just remove them from the file, because they contain the configuration, required in other apps.

If there is no idea how to hide them out of the box: Would it be possible to implement this feature for better compatibility with other todo.txt apps?

**Reply by gsantner** [reply]
For the view mode it is a option.

For edit mode not. It's a text editor, text is there = it is shown. Text is not there, so not shown. Showing **all** text is the editors job :-).

**Comment by aubreyz** [comment]
Just bear in mind this is not a proper todo.txt app. If you want a proper todo.txt app use something like simpletask. 

Markor is a great editor, but I tend to forget the todo part. May be useful on the odd occasion to track a specific project.

**Reply by gsantner** [reply]
> Just bear in mind this is not a proper todo.txt app

Sad to hear your view, I have strongly different opinion. I ditched all other todo.txt specific applications because Markor is superior to me. For todo.txt, but also Markdown :D.

**Reply by aubreyz** [reply]
I think it depends what you are doing. I have about 3000 tasks. When I am at work I need to see only the tasks that are related to work, that are passed the threshold date, and sorted by due date, then priority then start (threshold date). I don't want to see hidden tasks. And I want to be able to edit tasks from that view and further filter by a search from that view. 

Then I need to be able to shift to a different view at the push of a button. 

Markor may be good for smaller scale things, but it is not going to do anything like that.

**Reply by gsantner** [reply]
If you use todo.txt in this scale and mass you might just rethink if it's the right thing for the right thing. For everything there is a tool that fits more properly. Or you rethink how you organize and use the tools.

For the own good / personal mental health - I wouldn't mix and match work and personal todos. If it's like 1500/1500, split that up already to two todo.txt files. Everybody needs the time out of job.

> When I am at work I need to see only the tasks that are related to work

For this todo.txt have something covered, at it's core with contexts and projects. If it's about work, use `@work`. Easy as that. And Markor can filter to only show `@work`.

----

I don't really care how you want your flow, but I personally would think off using some bigger explicit todo & task management software, which is not using todo.txt. todo.txt is just what it is - a file containing tasks, one per line - and a lightweight format. Not more, not less.

**Comment by aisbergde** [comment]
I understand the idea to show in the editor all, what is there. This is a good idea. If I want to filter, I could just use other apps.

On the other side, the view mode has no syntax highlighting but shows only the poor text. So it is not helpful for todo.txt - at least for me. Or did I miss an option to get the reach visual features, which I see in the editor, also in the view mode?

Let me write about the editor mode. It is absolutely fine to show all content in the editor mode. I agree.

What about using special syntax highlight for rows, containing the string to mark the row as hidden?
Currently it is easy to visually detect tasks marked as done. Could we consider some special visual presentation, to still show them, but not so "present", as open tasks? But also nut strike through, like closed tasks.
And could this special meaning "hidden task" be used in the sorting? For example they could be at the very end.

I think, adding a specific syntax highlighting and using for sorting should be compatible to the current implementation?

**Reply by gsantner** [reply]
This is known yes - and needs somebody to work on. But your h:1 stuff would fit there. <https://github.com/gsantner/markor/issues/912>

**Comment by harshad1** [comment]
Alternatively, we could expand the filter system to enable some of this functionality. We already have the ability to create and save filters (tap on the title).

To (partially) fit @aubreyz workflow we would need to expand on the filter system to support
1. Threshold dates
3. AND and OR logic across filters

Internally, the filters use a simple json based syntax. We could expand on this and allow filtering based on a saved filter.

Building and maintaining this is far from trivial though.

**Comment by harshad1** [comment]
See filter syntax here: https://github.com/gsantner/markor/blob/master/app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtFilter.java#L95-L103
